google-cloud-firestore/samples/query_watch.rb

```diff
@@ -0,0 +1,199 @@
+# Copyright 2020 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+
+def listen_document project_id:, collection_path: "cities"
+  # project_id = "Your Google Cloud Project ID"
+  # collection_path = "cities"
+
+  firestore = Google::Cloud::Firestore.new project_id: project_id
+  # [START fs_listen_document]
+  doc_ref = firestore.col(collection_path).doc "SF"
+  snapshots = []
+
+  # Watch the document.
+  listener = doc_ref.listen do |snapshot|
+    # Process the snapshot.
+    puts "Received document snapshot: #{snapshot.document_id}"
+    snapshots << snapshot
+  end
+  # [END fs_listen_document]
+
+  sleep 1
+
+  # Create the document.
+  doc_ref.set(
+    name:       "San Francisco",
+    state:      "CA",
+    country:    "USA",
+    capital:    false,
+    population: 860_000
+  )
+  # Wait for the callback.
+  wait_until { snapshots.count == 1 }
+
+  # [START fs_detach_listener]
+  listener.stop
+  # [END fs_detach_listener]
+end
+
+def listen_changes project_id:, collection_path: "cities"
+  # project_id = "Your Google Cloud Project ID"
+  # collection_path = "cities"
+
+  firestore = Google::Cloud::Firestore.new project_id: project_id
+  # [START fs_listen_changes]
+  query = firestore.col(collection_path).where :state, :==, "CA"
```

Comment:
Should we drop the filter or pass it in?

Reply:
Sorry, I don't understand.

Reply:
I was wondering whether ` :state, :==, "CA"` should be a method argument here, but I can see that it shouldn't if this is the only code that gets added to the sample page.

Reply:
In this statement, `:state, :==, "CA"` are three positional arguments to `where`. The parentheses are omitted.